config/config.go

```diff
@@ -46,12 +69,13 @@ var legal = &struct {
 	documentStore: []string{"mongo"},
 }
 
+// TODO: This can be more perfomant by loading it just once and pass a copy for each call
 func New() (*config, error) {
-	log := logger.NewTag("config")
+	// log := logger.NewTag("config")
 	conf := new(config)
 
 	if err := viper.Unmarshal(conf); err != nil {
-		log.Error(e.FAIL, "err", err, "description", "Couldn't parse config")
+		slog.Error(e.FAIL, "err", err, "description", "Couldn't parse config")
```

Comment:
Aren't we doing the same thing in logger.go line 23 anyways? Why repeat it here..

Reply:
I'm afraid not, they are two different things. Please elaborate your concern?

Reply:
We are logging the error here, then again, in line 23 logger.go, we log the same error... We can just log it once ig?

Reply:
`config.New()` is one of the deepest call in the dependency chain. Meaning, `config` can be imported by any other package so it becomes critical that its errors are logged within its executing function rather than quietly handing over the error, and the responsibility to the calling function. In this case, the error pertains to a failed config Unmarshal attempt.

On the other hand, line 23 of logger.go is logging the error because the standard package log/slog's slog.New() does not return an error, and implicitly our logger.New() doesn't return an error as well. So, there's no way to pass the error to the calling function.